samza-core/src/main/java/org/apache/samza/serializers/model/JsonJobModelMixIn.java

```diff
@@ -23,11 +23,13 @@
 import org.apache.samza.config.Config;
 import org.apache.samza.job.model.ContainerModel;
 import org.codehaus.jackson.annotate.JsonCreator;
+import org.codehaus.jackson.annotate.JsonIgnoreProperties;
 import org.codehaus.jackson.annotate.JsonProperty;
 
 /**
  * A mix-in Jackson class to convert Samza's JobModel to/from JSON.
  */
+@JsonIgnoreProperties(ignoreUnknown = true)
```

Comment:
Do we need to add this annotation to `JsonTaskModelMixin` as well?

Reply:
@cameronlee314 Should we add them to other models as well? It makes migrations simpler if we add new properties in future. Specifically, allows rolling back to a version that does not have the newly added field.

Reply:
Good point. I'll add it to the others.

Reply:
It turns out that ContainerModel already has custom deserialization code which will ignore unknown properties (verified with unit test). So I only needed to add for TaskModel.